spec/unit/hanami/cli/commands/gem/new_spec.rb

```diff
@@ -66,437 +68,795 @@
+        expect(output).to include("Created app/assets/js/app.js")
+
+        # app/assets/css/app.css
+        app_css = <<~EXPECTED
+          body {
+            background-color: #fff;
+            color: #000;
+            font-family: sans-serif;
+          }
+        EXPECTED
+        expect(fs.read("app/assets/css/app.css")).to eq(app_css)
+        expect(output).to include("Created app/assets/css/app.css")
+
+        # app/assets/images/favicon.ico
+        expect(fs.exist?("app/assets/images/favicon.ico")).to be(true)
+
+        # app/db/relation.rb
+        relation = <<~EXPECTED
+          # frozen_string_literal: true
 
-          module Types
-            # Define your custom types here
+          require "hanami/db/relation"
+
+          module #{inflector.camelize(app)}
+            module DB
+              class Relation < Hanami::DB::Relation
+              end
+            end
           end
-        end
-      EXPECTED
-      expect(fs.read("lib/#{app}/types.rb")).to eq(types)
-      expect(output).to include("Created lib/bookshelf/types.rb")
+        EXPECTED
+        expect(fs.read("app/db/relation.rb")).to eq(relation)
+        expect(output).to include("Created app/db/relation.rb")
+
+        # app/db/struct.rb
+        struct = <<~EXPECTED
+          # frozen_string_literal: true
+
+          require "hanami/db/struct"
+
+          module #{inflector.camelize(app)}
+            module DB
+              class Struct < Hanami::DB::Struct
+              end
+            end
+          end
+        EXPECTED
+        expect(fs.read("app/db/struct.rb")).to eq(struct)
+        expect(output).to include("Created app/db/struct.rb")
+
+        expect(fs.read("app/structs/.keep")).to eq("")
+        expect(fs.read("app/repos/.keep")).to eq("")
+        expect(fs.read("config/db/migrate/.keep")).to eq("")
+
+        # lib/bookshelf/types.rb
+        types = <<~EXPECTED
+          # frozen_string_literal: true
 
-      # public/ error pages
-      expect(fs.read("public/404.html")).to include %(<title>The page you were looking for doesn’t exist (404)</title>)
-      expect(fs.read("public/500.html")).to include %(<title>We’re sorry, but something went wrong (500)</title>)
+          require "dry/types"
+
+          module #{inflector.camelize(app)}
+            Types = Dry.Types
+
+            module Types
+              # Define your custom types here
+            end
+          end
+        EXPECTED
+        expect(fs.read("lib/#{app}/types.rb")).to eq(types)
+        expect(output).to include("Created lib/bookshelf/types.rb")
+
+        # public/ error pages
+        expect(fs.read("public/404.html")).to include %(<title>The page you were looking for doesn’t exist (404)</title>)
+        expect(fs.read("public/500.html")).to include %(<title>We’re sorry, but something went wrong (500)</title>)
+      end
     end
   end
 
-  context "with head" do
-    let(:hanami_head) { true }
+  context "without hanami-db" do
```

Comment:
I wonder if this test could be made a little more meaningful and concise just by having it assert the absence of the DB files (or the absence of DB-related lines from other files), and then have it _presume_ that the rest of the app generation completes as expected?

(The downside is that the test would no longer catch if `--skip-db` causes breakages in the non-related files, but in the end, testing has to be about trade offs!)

Reply:
Definitely, great call.